src/Core/Scheduler/Scheduler.js

```diff
@@ -123,7 +123,13 @@ Scheduler.prototype.execute = function execute(command) {
     // parse host
     const layer = command.layer;
 
-    const host = layer.url ? new URL(layer.url).host : undefined;
+    let host;
+    try {
+        host = layer.url ? new URL(layer.url).host : undefined;
+    } catch (ex) {
+        host = undefined;
```

Comment:
We should at least test for the type of exception to only catch the one described in [the doc](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) and let other type of expression go up. 

Doing so is a good reflexe to have, because here we are catching everything, but other errors can happen that are unrelated to local file url problem (ex what if layer is null?). In these cases, we need to report these errors instead of assuming host = undefined.

Reply:
>  for example with local file paths

promises.push(globeView.addLayer({
    type: 'color',
    url: './ULTRA2009.gpx',
    protocol: 'rasterizer',
    id: 'Gpx',
}));

Reply:
I think `url` should always be a valid url. Either the user gives a valid string, or a valid url object (for instance `new URL('foo.gpx', window.location.origin + window.location.pathname)`) in `url`.

If you really want to allow strings like `./foo.gpx`, we could test if the string contains `://` and if not, construct the URL ourselves with the correct basepath, like just above. The idea is to avoid catching all exceptions here.

Reply:
Maybe we could simply always do: `new URL(layer.url, document.location)`?

Reply:
Indeed!

Reply:
thanks for your advice @autra @peppsac 
PR updated